### ISSUE TITLE
Memh request params on am_data_recv + multi lane assert on UCP_MAX_OP_MDS

### ISSUE THREAD

**Issue by petro-rudenko** [openucx/ucx#7979]
### Describe the bug
SparkUCX, setting memh request parameter to ucp_am_recv_data request causes assertion error.

### Steps to Reproduce
1. dt transport is used 
2. memh request parameter. W/o it it works.
3. memh was obtained from ucp_mem_map (Alloc flag)

```
#0  0x00007f97098a4efd in pause () from /lib64/libpthread.so.0
#1  0x00007f96ca2d4dee in ucs_debug_freeze () at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucs/debug/debug.c:818
#2  0x00007f96ca2d522e in ucs_error_freeze (message=0x7f96c96dc320 "Assertion `((sizeof(md_map) <= 4) ? __builtin_popcount((uint32_t)(md_map)) : __builtin_popcountl(md_map)) <= 4' failed") at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucs/debug/debug.c:921
#3  0x00007f96ca2d5848 in ucs_handle_error (message=0x7f96c96dc320 "Assertion `((sizeof(md_map) <= 4) ? __builtin_popcount((uint32_t)(md_map)) : __builtin_popcountl(md_map)) <= 4' failed") at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucs/debug/debug.c:1084
#4  0x00007f96ca2d200b in ucs_fatal_error_message (file=0x7f96c989c780 "/hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.c", line=477, function=0x7f96c989daf0 <__FUNCTION__.16511> "ucp_request_memory_reg_inner", message_buf=0x7f96c96dc320 "Assertion `((sizeof(md_map) <= 4) ? __builtin_popcount((uint32_t)(md_map)) : __builtin_popcountl(md_map)) <= 4' failed") at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucs/debug/assert.c:37
#5  0x00007f96ca2d217d in ucs_fatal_error_format (file=0x7f96c989c780 "/hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.c", line=477, function=0x7f96c989daf0 <__FUNCTION__.16511> "ucp_request_memory_reg_inner", format=0x7f96c989c4c5 "Assertion `%s' failed") at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucs/debug/assert.c:53
#6  0x00007f96c973ed5c in ucp_request_memory_reg_inner (uct_flags=0, req=0x7f96e44c4340, mem_type=UCS_MEMORY_TYPE_HOST, state=0x7f96e44c4398, datatype=8, length=50113, buffer=0x7f969e4fe000, md_map=151, context=0x7f9703fc1030) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.c:477
#7  ucp_request_memory_reg (context=0x7f9703fc1030, md_map=151, buffer=0x7f969e4fe000, length=50113, datatype=8, state=0x7f96e44c4398, mem_type=UCS_MEMORY_TYPE_HOST, req=0x7f96e44c4340, uct_flags=0) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.c:435
#8  0x00007f96c97d3718 in ucp_request_send_buffer_reg (uct_flags=0, md_map=151, req=0x7f96e44c4340) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.inl:517
#9  ucp_request_send_buffer_reg_lane_check (uct_flags=0, prev_md_map=23, lane=2 '\002', req=0x7f96e44c4340) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.inl:538
#10 ucp_send_request_add_reg_lane (lane=2 '\002', req=0x7f96e44c4340) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.inl:569
#11 ucp_rndv_progress_rma_zcopy_common (req=0x7f96e44c4340, lane=2 '\002', uct_rkey=90675350058861, proto=2) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:494
#12 0x00007f96c97e9017 in ucp_rndv_progress_rma_get_zcopy_inner (self=0x7f96e44c4420) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:1961
#13 ucp_rndv_progress_rma_get_zcopy (self=0x7f96e44c4420) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:1951
#14 0x00007f96c97d749b in ucp_request_try_send (req=0x7f96e44c4340) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.inl:334
#15 ucp_request_send (req=0x7f96e44c4340) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_request.inl:357
#16 ucp_rndv_req_send_rma_get (rndv_req=0x7f96e44c4340, rreq=0x7f96e44c3800, rndv_rts_hdr=0x7f96b5b3ce82, rkey_buf=0x7f96b5b3ceab) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:854
#17 0x00007f96c97ddeb8 in ucp_rndv_receive_inner (rkey_buf=0x7f96b5b3ceab, rndv_rts_hdr=0x7f96b5b3ce82, rreq=0x7f96e44c3800, worker=0x7f96c646d010) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:1515
#18 ucp_rndv_receive (worker=0x7f96c646d010, rreq=0x7f96e44c3800, rndv_rts_hdr=0x7f96b5b3ce82, rkey_buf=0x7f96b5b3ceab) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/rndv/rndv.c:1440
#19 0x00007f96c97194a7 in ucp_am_recv_data_nbx_inner (param=0x7f96c96dd810, count=50113, buffer=0x7f969e4fe000, data_desc=0x7f96b5b3ce82, worker=0x7f96c646d010) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_am.c:1146
#20 ucp_am_recv_data_nbx (worker=0x7f96c646d010, data_desc=0x7f96b5b3ce82, buffer=0x7f969e4fe000, count=50113, param=0x7f96c96dd810) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../src/ucp/core/ucp_am.c:1065
#21 0x00007f96c93db841 in Java_org_openucx_jucx_ucp_UcpWorker_recvAmDataNonBlockingNative (env=0x7f96e4215340, cls=0x7f96c96dd928, ucp_worker_ptr=140285548351504, data_descr_ptr=140285270281858, address=140284877856768, length=50113, callback=0x7f96c96dd940, request_params=0x7f96c96dd938) at /hpc/mtr_scrap/users/peterr/rapids/ucx/contrib/../bindings/java/src/main/native/worker.cc:345
```

error happens here: https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_request.inl#L537
```
prev_md_map = 23
ucp_ep_md_index(req->send.ep, lane) = 7
so new md_map = (1 << 7) | 23 = 151
popcount(151) = 5 > 4
```

From the log:
```
[1645551098.319398] [jazz10:187679:18]     ucp_request.c:459  UCX  REQ   req 0x7f96e44c4340: memh already registered
[1645551098.319406] [jazz10:187679:18]      dc_mlx5_ep.h:502  UCX  DEBUG iface 0x7f9703df5b70: allocate dci 0 for ep 0x7f96d8160dc0
[1645551098.319418] [jazz10:187679:18]     ucp_request.c:468  UCX  REQ   req 0x7f96e44c4340: mds 0x80 not registered - drop user memh
```

so it drops user memh, and doing rereg mds

**Comment by petro-rudenko** [comment]
Trasport doesn't matter here. Reproduces on rc. Only multilane. With a single lane it works.

**Comment by yosefe** [comment]
Fixed by #7987
